File: frontend/src/types/parametrage.ts
export interface Parametre {
  id: string;
  code: string;
  libelle: string;
  description?: string;
  type_parametre: 'GENERAL' | 'MODULE' | 'UTILISATEUR';
  module?: string;
  valeur: any;
  modifiable: boolean;
  visible: boolean;
  ordre: number;
  categorie?: string;
}

export interface ConfigurationModule {
  id: string;
  module: string;
  actif: boolean;
  configuration?: any;
  ordre_affichage: number;
  icone?: string;
  couleur?: string;
  roles_autorises: string[];
}

File: schemas/parametrage.py
from pydantic import BaseModel, UUID4
from typing import Optional, Dict, Any, List
from models.parametrage import TypeParametre, ModuleSysteme

class ParametreBase(BaseModel):
    code: str
    libelle: str
    description: Optional[str]
    type_parametre: TypeParametre
    module: Optional[ModuleSysteme]
    valeur: Dict[str, Any]
    modifiable: bool = True
    visible: bool = True
    ordre: int = 0
    categorie: Optional[str]

class ParametreCreate(ParametreBase):
    pass

class ParametreUpdate(ParametreBase):
    code: Optional[str]
    libelle: Optional[str]
    valeur: Optional[Dict[str, Any]]

class ParametreResponse(ParametreBase):
    id: UUID4

    class Config:
        orm_mode = True

class ConfigurationModuleBase(BaseModel):
    module: ModuleSysteme
    actif: bool = True
    configuration: Optional[Dict[str, Any]]
    ordre_affichage: int = 0
    icone: Optional[str]
    couleur: Optional[str]
    roles_autorises: List[str] = []

class ConfigurationModuleCreate(ConfigurationModuleBase):
    pass

class ConfigurationModuleUpdate(ConfigurationModuleBase):
    actif: Optional[bool]
    configuration: Optional[Dict[str, Any]]
    ordre_affichage: Optional[int]

class ConfigurationModuleResponse(ConfigurationModuleBase):
    id: UUID4

    class Config:
        orm_mode = True